src/storage-controller/src/collection_mgmt.rs

```diff
@@ -400,6 +415,19 @@ where
 
         Ok(MonotonicAppender::new(tx))
     }
+
+    fn get_read_only_rx(&self, id: GlobalId, force_writable: bool) -> watch::Receiver<bool> {
+        if *self.read_only_rx.borrow() && force_writable {
+            // In read-only mode we create a new shard for all migrated storage collections. So we
```

Comment:
Could you move this whole comment + assert + log line to where you set `force_writable` here https://github.com/MaterializeInc/materialize/pull/28643/files#diff-3d40535ccc11f976645c3c15ed4b6b1a08b7c04de0d21085ba2a4a891b6e5422R2744?

You might want to leave in the assert, just in case, though.

Reply:
Done.